src/main/java/com/microsoft/azure/functions/worker/WorkerLogManager.java

```diff
@@ -28,20 +29,33 @@ private void initializeImpl(JavaWorkerClient client, boolean logToConsole) {
         assert this.client == null && client != null;
         this.client = client;
         this.logToConsole = logToConsole;
-        addHostClientHandlers(this.hostLogger, null);
+        addHandlers(this.hostLogger, null);
     }
 
-    private void deinitializempl() {
+    private void deinitializImpl() {
         assert this.client != null;
         clearHandlers(this.hostLogger);
         this.logToConsole = false;
         this.client = null;
     }
 
+    private void flushLogsImpl() {
+        flush(emptyLogger);
+        flush(systemLogger);
+        flush(hostLogger);
+    }
+
+    private static void flush(Logger logger) {
+        Handler[] handlers = logger.getHandlers();
```

Comment:
it shouldn't have any issue, just want to double check for emptyLogger it won't have issue like NPE.

Reply:
No, verified this will not throw a null pointer